File: Java Code/Youtube_Link_Finder.java
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Youtube_Link_Finder {
	
	public static void main(String[] args) {
		String link_str;
		Scanner sc =new Scanner(System.in);
		try {
			link_str=sc.nextLine().trim();
			//link_str="https://youtu.be/KOOJDAO456w";
			//link_str="http://www.youtube.com/watch?v=KOOJDAO456w";
			String reg=".+[/|?v=]";
			Pattern pat=Pattern.compile(reg);
			Matcher mat=pat.matcher(link_str);
			if (mat.find()) {
				//int ind=mat.end();
				System.out.println(link_str.substring(mat.end()));
			}
;		} finally {
			sc.close();
		}
	}

}

File: Python/Youtube link Finder.py
import re
reg=re.compile(r'^(\w)+://(w{3}.)?youtu[.be|be.com][/(\w)+|/(\w)+?=(\w)+]')
#linkstr= input('Link: ').strip()
linkstr='https://www.youtube.com/watch?v=kbx365'
linkstr='https://youtu.be/kbx365'
if len(linkstr) >0:
    m=reg.search(linkstr)
    if m:
        links=m.group()
        print(links)
    else:
        print('Not Found')
else:
    print('Nothing')